lib/src/Uncertainty/Distribution/HistogramFactory.cxx

```diff
@@ -133,6 +140,63 @@ Histogram HistogramFactory::buildAsHistogram(const Sample & sample,
   return result;
 }
 
+Histogram HistogramFactory::buildAsHistogram(const Sample & sample,
+    const Scalar first,
+    const Point width) const
+{
+  const UnsignedInteger size = sample.getSize();
+  if (size == 0) throw InvalidArgumentException(HERE) << "Error: cannot build an Histogram based on an empty sample.";
+  if (sample.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: can build an Histogram only if dimension equals 1, here dimension=" << sample.getDimension();
+  const UnsignedInteger binNumber = width.getSize();
+  if (binNumber == 0) throw InvalidArgumentException(HERE) << "Error: expected a positive number of bin, got 0.";
+  // Construct the histogram
+  Point heights(binNumber, 0.0);
+  // It will extends from min to max.
+  const Scalar minvalue = sample.getMin()[0];
+  const Scalar maxvalue = sample.getMax()[0];
+  if (!SpecFunc::IsNormal(minvalue) || !SpecFunc::IsNormal(maxvalue)) throw InvalidArgumentException(HERE) << "Error: cannot build an Histogram distribution if data contains NaN or Inf";
+  if (maxvalue == minvalue)
+  {
+    const Scalar epsilon = ResourceMap::GetAsScalar("Distribution-DefaultCDFEpsilon");
+    const Scalar delta = std::max(std::abs(minvalue), 10.0) * epsilon;
+    Histogram result(minvalue - 0.5 * delta, Point(1, delta), Point(1, 1.0));
+    result.setDescription(sample.getDescription());
+    return result;
+  }
+  if (minvalue<first) throw InvalidArgumentException(HERE) << "Error: the minimum of the sample is smaller than first";
+  // Compute the right bound of the histogram = first + the sum of the widths
+  Scalar rightBound = first;
+  for(UnsignedInteger j = 0; j < binNumber; ++j)
+  {
+    rightBound += width[j];
+  }
+  if (maxvalue>rightBound) throw InvalidArgumentException(HERE) << "Error: the maximum of the sample is greater than the right boundary of the histogram";
+  // Count the number of points in each class with a naive loop
+  Scalar left = first;
+  Scalar right;
+  for(UnsignedInteger j = 0; j < binNumber; ++j) // Loop over the classes
+  {
+    right = left + width[j];
+    for(UnsignedInteger i = 0; i < size; ++i)    // Loop over the values
+    {
+      if ((sample(i,0) >= left) and (sample(i,0)<right))
```

Comment:
no 'and', use &&

Reply:
With this test, if a point is exactly equal to maxvalue it is not taken into account

Reply:
Ok for the "and". 
For the sample(i,0)==maxvalue case, I understand your point, but what you wrote is probably not what you meant. In the current implementation, there is an issue if the maxvalue is exactly equal to rightBound. In order to manage this case, I change the condition "if (maxvalue > rightBound)" to "(maxvalue >= rightBound)". This fixes the issue.